lib/game.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require_relative 'board.rb'
+require_relative 'player.rb'
+# Game class
+class Game
+  include UI
+
+  attr_accessor :continue
+  def initialize(board, player1, player2)
+    @new_board = board
+    @player1 = player1
+    @player2 = player2
+    play
+  end
+
+  def play
+    while @new_board.not_full?
+      playing(@player1, @new_board)
+      return if is_finished?(@new_board)
+
+      playing(@player2, @new_board)
+      return if is_finished?(@new_board)
+    end
+  end
+
+  private
+
+  def playing(player, board)
```

Comment:
Let's further simplify this method guys. Benefits of simplifying this further:

- easier to extend the capability of your app
- easier to debug should things not work (during capability extension or refactoring)

Right now the method is still doing a lot. I suggest simplifying your logic as such:

```ruby
class Game

initialize(board, player1, player2)
  # initialize board, player1 and player2 variables
end

def play
  until game_is_over?
    # 1. call player1's make_move method unless has_won?(@player2)
    # 2. call player2's make_move method unless has_won?(@player1)
   end
end

private

def game_is_over?
  has_won?(@player1) || has_won?(@player2) || !@board.not)full?
end

def has_won?(player)
 # in this method you will check if one of the arrays inside `@winner_points` in `@board` has been filled with the player's symbol/sign
end
```